packages/dx-react-grid-bootstrap3/src/templates/filter-row/filter-selector.jsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { ListGroup, ListGroupItem } from 'react-bootstrap';
+import { Overlay } from '../parts/overlay';
+
+export class FilterSelector extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = { opened: false };
+
+    this.handleButtonClick = () => {
+      this.setState({ opened: !this.state.opened });
+    };
+    this.handleOverlayHide = () => {
+      this.setState({ opened: false });
+    };
+    this.handleMenuItemClick = (nextValue) => {
+      this.setState({ opened: false });
+      this.props.onChange(nextValue);
+    };
+
+    this.getTargetElement = () => this.targetElement;
+  }
+  render() {
+    const {
+      value, availableValues, disabled, getMessage, iconComponent: Icon,
+    } = this.props;
+    const { opened } = this.state;
+
+    return availableValues.length ? (
+      <span className="input-group-btn">
+        <button
+          className="btn btn-default"
+          disabled={disabled || availableValues.length === 1}
+          onClick={this.handleButtonClick}
+          ref={(ref) => { this.targetElement = ref; }}
+        >
+          <Icon type={value} />
+        </button>
+        <Overlay
+          visible={opened}
+          target={this.getTargetElement()}
```

Comment:
Just a question: why we create the `getTargetElement` function instead `this.targetElement` use?